### ISSUE TITLE
Suggestion: Multi-tag searching/refinement

### ISSUE THREAD

**Issue by mellamokb** [face-it/Hangfire.Tags#28]
Since you are so amazingly quick to implement suggestions, here is another one :)

Just to give some background, we operate on a scale of about 50,000 jobs per week.  At any point in time we have around 15-20k jobs in the Scheduled status, going from a few minutes to months in the future.  Also I configured to hold on to completed jobs for up to 30 days for basic troubleshooting.  That means in the complete job status we have typically around 1.5m jobs available for searching. Hence the use of Hangfire Tags is immense to help searching!  (Before I discovered this I was using a bunch of handcrafted queries directly on the DB tables/json values to find and sort jobs)

To get a starting point, I made every job type (method name) a separate tag, and then every parameter value as a unique prefixed tag.  For example, 'employeeid:123', or 'companycode:abcc' (yeah I realized afterward the ':' gets stripped out but it's okay).  That way not only does every parameter have a unique searchable tag, but I can tell two int parameter value types apart.  Unfortunately after only a week of production usage, I already have 41k tags, but I have not seen any performance issue thankfully!  I also use the dropdown method instead of tag cloud.

Now to the question.  When I am searching by tag, I often want to look for a combination of method, and one or more parameter values.  Would it be possible for the tag search interface to allow entering multiple tags at a time, and search for the combination of all jobs having all entered tags?  And maybe to go with that, when you are on the search results by tag, could there be a way to further refine the search, by entering more tags, but filtered only for jobs that match the existing tag filter, if that makes sense?

Thanks a bunch for all your efforts!!

**Edit** I don't know if there is a DM ability in Github, but I'd be willing to make a donation to your efforts for this request, let me know.  Thanks again!

**Comment by erwin-faceit** [comment]
Hey, thanks for the idea. Actually, it's partially implemented, it's just the interface that is missing. You can already filter it by yourself by seperating tags in the url with a comma. It will do an AND search in that case, which seems to be exactly what you want.

In addition to your tags, I also have a DailyTag added globally, which will add the date as a tag, so I can easily search for a specific job on a specific day. 

Let me know if this helps!

**Comment by erwin-faceit** [comment]
I added a tag search refinement on the page with the jobs, that belong to the specified tag. So you first select an initial tag, which will take you to the results. On that page you can select another tag, which is somehow related to the first tag. The relation between the tags is based on the jobs that share the same tags. You should update to version 1.7.0 to get this feature, and it's only enabled when using the DropDown style.

**Comment by mellamokb** [comment]
Hey, just wanted to note I am very pleased with how the multi tag works.  I had discovered early on about the comma-separated list in the url and had been using that prior to search jobs.

As I mentioned in the other case we have millions of tagged jobs, and 10s of thousands of tags.  It is wonderful to be able to find the job I'm looking for when we need to troubleshoot a customer issue.  Before all this I had to wade through the tables manually with custom queries and it would sometimes take hours to find one job.  Thanks again for your efforts :)
